snuba/datasets/processors/transactions_processor.py

```diff
@@ -472,4 +474,10 @@ def process_message(
         # the following operation modifies the event_dict and is therefore *not* order-independent
         self._process_contexts_and_user(processed, event_dict)
 
-        return InsertBatch([processed], None)
+        try:
+            raw_received = _collapse_uint32(int(event_dict["data"]["received"]))
+            assert raw_received is not None
+            received = datetime.utcfromtimestamp(raw_received)
+            return InsertBatch([processed], received)
+        except KeyError:
+            raise KeyError("Missing received timestamp field in transaction")
```

Comment:
Wasn't the point of the try/except to fall back to the old behavior in case we hit the KeyError?

Reply:
Just to be clear: today, what happens if there is a payload missing the `received` field is that the `MessageProcessor` will just drop it - is this correct?

So I guess we can fall back to the old behavior, but we also want to see the error show up in Sentry so we can know if there were any payloads missing the field. What's the right way of getting that error to show up?

Reply:
Alright I can do that then, thanks